### PR TITLE
readme/faq: Add info on VSS alternatives

### PR DESCRIPTION
**Pull request by dsprenkels**
This PR adds two questions to the FAQ, about when *not* to use SSS.

I did not have this exact question from anybody, but there seem to exist a lot of folks (mostly blockchainers) that think Shamir secret sharing is the silver bullet of secret sharing schemes. While it is very easy and produces small shares (and fast because we do not need group arithmetic), some cases really need a verifiable secret sharing scheme. These questions try to give some insight into when a VSS scheme would probably be better.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -196,7 +196,48 @@ actually how [sss-cli] produces variable-length shares.)
...
-Feel free to send me an email on my Github associated e-mail address.
+### I do not know a lot about secret sharing. Is Shamir secret sharing useful for me?
+
+It depends. In the case of threshold schemes (that's what this is) there are
+two types:
+
+1. The share-holders can verify that their shares are valid.
+2. The share-holders can _not_ verify that their shares are valid.
+
+Shamir's scheme is of the first type. This immediately implies that the dealer
```
Comment (yorickvP) [inline README.md]: second?

Comment (dsprenkels) [reply README.md]: Thanks! Fixed it.

File: README.md [hunk 2]
```diff
@@ -196,7 +196,48 @@ actually how [sss-cli] produces variable-length shares.)
 
 ## Questions
 
-Feel free to send me an email on my Github associated e-mail address.
+### I do not know a lot about secret sharing. Is Shamir secret sharing useful for me?
+
+It depends. In the case of threshold schemes (that's what this is) there are
```
Comment (mrngm) [inline README.md]: Since it's not mentioned elsewhere, perhaps add a single line explaining _threshold schemes_?

File: README.md [hunk 3]
```diff
@@ -196,7 +196,48 @@ actually how [sss-cli] produces variable-length shares.)
...
+two types:
+
+2. The share-holders _cannot_ verify that their shares are valid.
+1. The share-holders _can_ verify that their shares are valid.
+
+Shamir's scheme is of the first type. This immediately implies that the dealer
+could cheat. Indeed, they can distribute a number of shares which are are just
```
Comment (Minnozz) [inline README.md]: `are are` (duplicate word)

File: README.md (hunk 2 above)
Comment (dsprenkels) [reply README.md]: Hmmm. Maybe leave out the term in total? I do not see explaining it here having a lot of benefit to the reader.

File: README.md (hunk 3 above)
Comment (dsprenkels) [reply README.md]: Thanks!

File: README.md (hunk 2 above)
Comment (dsprenkels) [reply README.md]: Nevermind, we mention it in the introduction. :P

**Pull request merged**